subprojects/base-services/src/main/java/org/gradle/internal/Pair.java

```diff
@@ -18,33 +18,39 @@
 
 import com.google.common.base.Function;
 
+import javax.annotation.Nullable;
+
 public final class Pair<L, R> {
 
     public final L left;
     public final R right;
 
-    private Pair(L left, R right) {
+    private Pair(@Nullable L left, @Nullable R right) {
```

Comment:
❓ Is this constructor ever passed `null` values?

Reply:
Yes from `ProjectPropertySettingBuildLoader` via `Pair.of(l,r)`.

Reply:
Ok.